cluster/images/etcd/migrate/options.go

```diff
@@ -0,0 +1,242 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/klog/v2"
+)
+
+var (
+	supportedEtcdVersions = []string{"3.0.17", "3.1.12", "3.2.24", "3.3.17", "3.4.9"}
+)
+
+const (
+	etcdNameEnv           = "ETCD_NAME"
+	etcdHostnameEnv       = "ETCD_HOSTNAME"
+	hostnameEnv           = "HOSTNAME"
+	dataDirEnv            = "DATA_DIRECTORY"
+	initialClusterEnv     = "INITIAL_CLUSTER"
+	initialClusterFmt     = "%s=http://localhost:%d"
+	peerListenUrlsEnv     = "LISTEN_PEER_URLS"
+	peerListenUrlsFmt     = "http://localhost:%d"
+	peerAdvertiseUrlsEnv  = "INITIAL_ADVERTISE_PEER_URLS"
+	peerAdvertiseUrlsFmt  = "http://localhost:%d"
+	targetVersionEnv      = "TARGET_VERSION"
+	targetStorageEnv      = "TARGET_STORAGE"
+	etcdDataPrefixEnv     = "ETCD_DATA_PREFIX"
+	etcdDataPrefixDefault = "/registry"
+	ttlKeysDirectoryFmt   = "%s/events"
+	etcdServerArgsEnv     = "ETCD_CREDS"
+)
+
+type migrateOpts struct {
+	name              string
+	port              uint64
+	peerPort          uint64
+	peerListenUrls    string
+	peerAdvertiseUrls string
+	binDir            string
+	dataDir           string
+	bundledVersions   []string
+	supportedVersions SupportedVersions
+	etcdDataPrefix    string
+	ttlKeysDirectory  string
+	initialCluster    string
+	targetVersion     string
+	targetStorage     string
+	etcdServerArgs    string
+}
+
+func parseFlags(flags *flag.FlagSet, opt *migrateOpts) {
+	// FIXME:
```

Comment:
When we agree that this approach makes sense, I will update helps for each flag to reflect fallback env vars.
@jpbetz